client/service/src/lib.rs

```diff
@@ -160,6 +160,7 @@ async fn build_network_future<
 	network: sc_network::NetworkWorker<B, H>,
 	client: Arc<C>,
 	sync_service: Arc<SyncingService<B>>,
+	network_block: Arc<Box<dyn NetworkBlock<B::Hash, NumberFor<B>> + Send + Sync>>,
```

Comment:
The custom block syncing service should be offered by `SyncingService` since it implements `NetworkBlock`. It doesn't make sense to me to pass another syncing-related handle that only deals with `NetworkBlock`. If `SyncingService` doesn't work with whatever your implementation requires, could you please open a discussion instead so we could try and work out how Substrate could possibly provide that without having to use these kinds of workarounds. We have just started refactoring the syncing code so it would be nice to hear your thoughts on what's wrong with the current implementation.

I also don't like that it's `Arc<Box<dyn ...>>` but I assume that is because Rust can't downcast `SyncingService` to `NetworkBlock`?

Reply:
Opened https://github.com/paritytech/substrate/discussions/14243

`Box` is needless, I made it unnecessarily complex for no reason somehow :P

Reply:
@altonen Updated PR, please take another look.